lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -217,7 +220,9 @@ def update_action_builder(args, source)
         end
         source['script']['lang'] = @options[:script_lang] if @options[:script_lang] != ''
       else
-        source = { 'doc' => source }
+        unless source.has_key?('doc')
```

Comment:
Yes. It's probably better to not mix two non related issues in the same PR. I think there is another solution for #356 without breaking change like this !